### PR TITLE
handle noise messages

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Adds messages captured under gpsdd_kilifi_s01_noise_handler_activation flow.
Adds a script to export contacts with messages labelled with NOC code for textit.

### CODE REVIEW

File: export_noise_contacts.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    noise_uuids = set()
+    skipped = 0
+    for path in traced_data_paths:
+        # Load the traced data
+        log.info(f"Loading previous traced data from file '{path}'...")
+        with open(path) as f:
+            data = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
+        log.info(f"Loaded {len(data)} traced data objects")
+
+        # Filter for participants that opted-in via the close-out.
```
Comment (as2388) [inline export_noise_contacts.py]:
```suggestion
        # Filter for participants that are NOC in any field
```

File: export_noise_contacts.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        firestore_uuid_table_credentials,
+        "avf-phone-uuid-"
+    )
+    log.info("Initialised the Firestore UUID table")
+
+    noise_uuids = set()
+    skipped = 0
+    for path in traced_data_paths:
+        # Load the traced data
+        log.info(f"Loading previous traced data from file '{path}'...")
```
Comment (as2388) [inline export_noise_contacts.py]: Does this work? Some projects used to filter noise participants, so I can't remember if they were guaranteed to make it to the individuals traced data

File: export_noise_contacts.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.DEMOG_CODING_PLANS:
+                for cc in plan.coding_configurations:
+                    if cc.coding_mode == CodingModes.SINGLE:
+                        codes = [cc.code_scheme.get_code_with_code_id(td[cc.coded_field]["CodeID"])]
+                    else:
+                        assert cc.coding_mode == CodingModes.MULTIPLE
+                        codes = [cc.code_scheme.get_code_with_code_id(label["CodeID"]) for label in td[cc.coded_field]]
+
+                    for code in codes:
+                        if code.string_value == "NOC":
```
Comment (as2388) [inline export_noise_contacts.py]:
```suggestion
                        if code.string_value == Codes.NOISE_OTHER_CHANNEL:
```

**Review by as2388 — APPROVED** [review]
Thanks @IsaackMwenda, sorry you're facing this problem. Approved under the assumption that you didn't add an NOC filter in the pipelines.

File: export_noise_contacts.py (hunk 1 above)
Comment (IsaackMwenda) [reply export_noise_contacts.py]: updated thanks.

File: export_noise_contacts.py (hunk 2 above)
Comment (IsaackMwenda) [reply export_noise_contacts.py]: updated to load one traced data file.

File: export_noise_contacts.py (hunk 3 above)
Comment (IsaackMwenda) [reply export_noise_contacts.py]: updated thanks

**Pull request merged**